includes/create-theme/theme-patterns.php

```diff
@@ -21,6 +21,30 @@ public static function pattern_from_template( $template, $new_slug = null ) {
 		);
 	}
 
+	public static function pattern_from_wp_block( $pattern_post ) {
+		$pattern               = new stdClass();
+		$pattern->id           = $pattern_post->ID;
+		$pattern->title        = $pattern_post->post_title;
+		$pattern->name         = sanitize_title_with_dashes( $pattern_post->post_title );
+		$pattern->slug         = wp_get_theme()->get( 'TextDomain' ) . '/' . $pattern->name;
+		$pattern_category_list = get_the_terms( $pattern->id, 'wp_pattern_category' );
+		$pattern->categories   = ! empty( $pattern_category_list ) ? join( ', ', wp_list_pluck( $pattern_category_list, 'name' ) ) : '';
+		// $pattern->sync_status  = get_post_meta( $pattern->id, 'wp_pattern_sync_status', true );
```

Comment:
I'm not sure this belongs here but I've left it here commented out as a reference for now.